src/renderers/shared/stack/event/ReactSyntheticEventType.js

```diff
@@ -8,14 +8,29 @@
  *
  * Flow type for SyntheticEvent class that includes private properties
  *
- * @providesModule ReactSyntheticEvent
+ * @providesModule ReactSyntheticEventType
  * @flow
  */
 
 'use strict';
 
-export type DispatchConfig = any;
+import type {ReactInstance} from 'ReactInstanceType';
+
+export type DispatchConfig = {
+  dependencies: Array<string>,
+  phasedRegistrationNames?: {
+    bubbled: string,
+    captured: string,
+  },
+  registrationName?: string,
+};
 
 export class ReactSyntheticEvent extends SyntheticEvent {
   dispatchConfig: DispatchConfig;
+  getPooled: (
```

Comment:
I typed PooledClass so you may be able to use it and it's going to get typed:

https://github.com/facebook/react/blob/a70acb37d9dc392b395bd921d90d5488542c2402/src/shared/utils/PooledClass.js#L105-L111
https://github.com/facebook/react/blob/a70acb37d9dc392b395bd921d90d5488542c2402/src/renderers/shared/utils/CallbackQueue.js#L106

The downside of my version is that it ignores the arguments of the constructor.

On a tangent, I think we should stop using definitions that are shipped with flow for the internal codebase and duplicate them for now. When they are good enough inside of React then have React ship a react.flow.js file based on our definitions and remove them from flow altogether.